internal/comm/httptransport_test.go

```diff
@@ -425,40 +441,45 @@ func TestHTTPTransport_UpdatePeers(t *testing.T) {
 	defer tr2.Close()
 
 	// server 3 starts on a new port, not reflected yet in shared config.
-	// servers 1,2 will not be able to reach it.
+	// servers 1,2 will not be able to reach it. However, server 3 can reach servers 1,2.
 	localConfigs[2].Replication.Network.Port++
 	tr3, cl3, err := startTransportWithLedger(t, lg, localConfigs, sharedConfig, 2, 5)
 	require.NoError(t, err)
 	defer tr3.Close()
 
-	// messages are not received
-	err = tr1.SendConsensus([]raftpb.Message{{To: 3}})
-	require.NoError(t, err)
-	require.Eventually(t,
-		func() bool {
-			return cl1.ReportUnreachableCallCount() == 1
-		},
-		10*time.Second, 10*time.Millisecond,
-	)
-	err = tr2.SendConsensus([]raftpb.Message{{To: 3}})
-	require.NoError(t, err)
-	err = tr2.SendConsensus([]raftpb.Message{{To: 3}})
-	require.NoError(t, err)
-	require.Eventually(t,
-		func() bool {
-			return cl2.ReportUnreachableCallCount() == 2
-		},
-		10*time.Second, 10*time.Millisecond,
-	)
+	wg.Wait()
+	time.Sleep(1 * time.Second)
```

Comment:
I don't like unconditional sleep. Is any way to make it eventually?

Reply:
Done